data-processing/explanations/colorize_tex.py

```diff
@@ -145,7 +148,10 @@ def colorize_citations(tex: str) -> Iterator[CitationColorizationBatch]:
             hue_generator = generate_hues()
 
             # And get the hue for the next entity.
-            hue = next(hue_generator)
+            if preset_hue is None:
+                hue = next(hue_generator)
+            else:
+                hue = preset_hue
```

Comment:
Technically this condition isn't necessary as this exception will never be reached if preset_hue is reached. Just wanted to point that out but I'm good with it there if you want it there for explicitness